tests/components/insteon/test_config_flow.py

```diff
@@ -0,0 +1,703 @@
+
+    result = await _import_config(hass, MOCK_IMPORT_FULL_CONFIG_HUB_V1)
+
+    assert result["type"] == "create_entry"
+    assert hass.config_entries.async_entries(DOMAIN)
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        assert entry.data[CONF_HOST] == MOCK_HOSTNAME
+        assert entry.data[CONF_PORT] == MOCK_PORT
+        assert entry.data[CONF_HUB_VERSION] == 1
+        assert CONF_USERNAME not in entry.data
+        assert CONF_PASSWORD not in entry.data
+        assert CONF_OVERRIDE not in entry.data
+        assert CONF_X10 not in entry.data
+        assert entry.options[CONF_OVERRIDE][0][CONF_ADDRESS] == "1A.2B.3C"
+        assert entry.options[CONF_X10][0][CONF_HOUSECODE] == MOCK_HOUSECODE
+
+
+async def test_import_full_hub_v2(hass: HomeAssistantType):
+    """Test importing a full Hub v2 config from yaml."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    result = await _import_config(hass, MOCK_IMPORT_FULL_CONFIG_HUB_V2)
+
+    assert result["type"] == "create_entry"
+    assert hass.config_entries.async_entries(DOMAIN)
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        assert entry.data[CONF_HOST] == MOCK_HOSTNAME
+        assert entry.data[CONF_PORT] == MOCK_PORT
+        assert entry.data[CONF_USERNAME] == MOCK_USERNAME
+        assert entry.data[CONF_PASSWORD] == MOCK_PASSWORD
+        assert entry.data[CONF_HUB_VERSION] == 2
+        assert CONF_OVERRIDE not in entry.data
+        assert CONF_X10 not in entry.data
+        assert entry.options[CONF_OVERRIDE][0][CONF_ADDRESS] == "1A.2B.3C"
+        assert entry.options[CONF_X10][0][CONF_HOUSECODE] == MOCK_HOUSECODE
+
+
+async def test_import_min_hub_v2(hass: HomeAssistantType):
+    """Test importing a minimum Hub v2 config from yaml."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    result = await _import_config(hass, MOCK_IMPORT_MINIMUM_HUB_V2)
+
+    assert result["type"] == "create_entry"
+    assert hass.config_entries.async_entries(DOMAIN)
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        assert entry.data[CONF_HOST] == MOCK_HOSTNAME
+        assert entry.data[CONF_PORT] == 25105
+        assert entry.data[CONF_USERNAME] == MOCK_USERNAME
+        assert entry.data[CONF_PASSWORD] == MOCK_PASSWORD
+        assert entry.data[CONF_HUB_VERSION] == 2
+
+
+async def test_import_min_hub_v1(hass: HomeAssistantType):
+    """Test importing a minimum Hub v1 config from yaml."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    result = await _import_config(hass, MOCK_IMPORT_MINIMUM_HUB_V1)
+
+    assert result["type"] == "create_entry"
+    assert hass.config_entries.async_entries(DOMAIN)
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        assert entry.data[CONF_HOST] == MOCK_HOSTNAME
+        assert entry.data[CONF_PORT] == 9761
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)